### PR TITLE
Resolves return_img for Plotter.show()

### PR DESCRIPTION
**Pull request by beroda**
### Overview

Resolves https://github.com/pyvista/pyvista/issues/3241

### Details

Now, `Plotter.show()` with `return_img = True` works even if `return_cpos = False`.

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
Sorry for the poor dev experience, but our image cache regression testing is currently broken.  After we fix it, hopefully in #3748, you will have to add in images to the cache.

We shouldn't merge this until #3748 is merged.

File: pyvista/plotting/plotting.py
Comment (banesullivan) [inline pyvista/plotting/plotting.py:6241]:
```suggestion
            return self.last_image
```

File: tests/plotting/test_plotting.py
Comment (banesullivan) [inline tests/plotting/test_plotting.py:339]: As @MatthewFlamm pointed out, our image regression testing is currently broken, but these tests will likely need to add images for the cache as I believe the default is for all tests in `test_plotting.py` to use the cache... I think...

**Review by banesullivan — CHANGES_REQUESTED** [review]
Awesome! Thank you for fixing this @beroda! 

We can handle adding the image regression testing nonsense here, so if this sounds foreign to you, don't worry about it!

**Comment by beroda** [comment]
Not sure if the tests I implemented for this PR require to be compared to cached images. They only verify that the return statements in Plotter.show works as expected.

**Comment by banesullivan** [comment]
> Not sure if the tests I implemented for this PR require to be compared to cached images. They only verify that the return statements in Plotter.show works as expected.

Your absolutely correct. However since these tests are placed in the `tests/plotting/test_plotting.py` module, image regression testing is automatic. Perhaps a better place for these tests would be in `tests/test_plotter.py`

File: tests/plotting/test_plotting.py
Comment (banesullivan) [reply tests/plotting/test_plotting.py:339]: See PR comment about moving to `tests/test_plotter.py`

File: pyvista/plotting/plotting.py
Comment (banesullivan) [reply pyvista/plotting/plotting.py]: I suppose this is purely stylistic

File: tests/plotting/test_plotting.py
Comment (beroda) [reply tests/plotting/test_plotting.py:339]: Sounds better. I will move them asap.

Comment (beroda) [reply tests/plotting/test_plotting.py]: Done

**Review by MatthewFlamm — APPROVED** [review]
I no longer think we need to wait for the other PR first since this no longer needs image cache testing.

**Review by banesullivan — APPROVED** [review]

**Pull request merged**
